wiki/start/a-simple-game.md

```diff
@@ -5,605 +5,1091 @@ redirect_from:
-      // a sound effect as well.
-      for (Iterator<Rectangle> iter = raindrops.iterator(); iter.hasNext(); ) {
-         Rectangle raindrop = iter.next();
-         raindrop.y -= 200 * Gdx.graphics.getDeltaTime();
-         if(raindrop.y + 64 < 0) iter.remove();
-         if(raindrop.overlaps(bucket)) {
-            dropSound.play();
-            iter.remove();
-         }
-      }
-   }
-
-   @Override
-   public void dispose() {
-      // dispose of all the native resources
-      dropImage.dispose();
-      bucketImage.dispose();
-      dropSound.dispose();
-      rainMusic.dispose();
-      batch.dispose();
-   }
-}
-```
-
-## Where to go from here
-This was a very basic example of how to use libGDX to create a minimalistic game. There are quite a few things that can be improved from here on. Your next steps should most certainly entail looking at `Screen`s and `Game`s. To learn about these, there is a **[second tutorial](/wiki/start/simple-game-extended)** following on from this one.
+import com.badlogic.gdx.utils.viewport.FitViewport;
+
+public class Main implements ApplicationListener {
+    Texture backgroundTexture;
+    Texture bucketTexture;
+    Texture dropTexture;
+    Sound dropSound;
+    Music music;
+    SpriteBatch spriteBatch;
+    FitViewport viewport;
+    Sprite bucketSprite;
+    Vector3 touchPos;
+    Array<Sprite> dropSprites;
+    long lastDropTime;
+    Rectangle bucketRectangle;
+    Rectangle dropRectangle;
+
+    @Override
+    public void create() {
+        backgroundTexture = new Texture("background.png");
+        bucketTexture = new Texture("bucket.png");
+        dropTexture = new Texture("drop.png");
+        dropSound = Gdx.audio.newSound(Gdx.files.internal("drop.mp3"));
+        music = Gdx.audio.newMusic(Gdx.files.internal("music.mp3"));
+        viewport = new FitViewport(8, 5);
+        spriteBatch = new SpriteBatch();
+        bucketSprite = new Sprite(bucketTexture);
+        bucketSprite.setSize(1, 1);
+        touchPos = new Vector3();
+        dropSprites = new Array<>();
+        bucketRectangle = new Rectangle();
+        dropRectangle = new Rectangle();
+        music.setLooping(true);
+        music.setVolume(.5f);
+        music.play();
+    }
+
+    @Override
+    public void resize(int width, int height) {
+        viewport.update(width, height, true);
+    }
+
+    @Override
+    public void render() {
+        input();
+        logic();
+        draw();
+    }
+
+    private void input() {
+        float delta = Gdx.graphics.getDeltaTime();
+
+        if (Gdx.input.isKeyPressed(Keys.RIGHT)) {
+            bucketSprite.setX(bucketSprite.getX() + 4f * delta);
+        } else if (Gdx.input.isKeyPressed(Keys.LEFT)) {
+            bucketSprite.setX(bucketSprite.getX() - 4f * delta);
+        }
+
+        if (Gdx.input.isTouched()) {
+            touchPos.set(Gdx.input.getX(), Gdx.input.getY(), 1);
+            viewport.unproject(touchPos);
+            bucketSprite.setX(touchPos.x);
```

Comment:
Subtracting half its width here would feel nicer.

Reply:
Included. Again, didn't even know this was a method. I'm so deep into my own way of doing stuff.